### ISSUE TITLE
Java: Extractor succeeds for `codeql test run` despite compilation errors in source file

### ISSUE THREAD

**Issue by Marcono1234** [github/codeql#5476]
**Version**
CodeQL CLI v2.4.6

**Description of the issue**
(Reported this for github/codeql instead of github/codeql-cli-binaries since this might be specific to the Java extractor.)

It appears the Java extractor executed as part of `codeql test run` succeeds despite the test `.java` source files containing compilation errors. When running with `--show-extractor-output -vvvv` it even logs that it is running with `--strict-javac-errors`, but still does not fail.

**Reproduction steps**
Assumes you already have a project with the Java library present; you could for example use https://github.com/github/codeql and perform the following in the test directory of it.

1. Create a test directory with the following `Test.java` file:
    ```java
    class Test {
        String test() {
            // Compilation error: missing return
        }

        void test2() {
            while (true) { }

            // Compilation error: unreachable
            System.out.println("test");
        }
    }
    ```
    As indicated by the comments, the file causes compilation errors which can be seen when running `javac`.
2. Create a `Test.ql` file (content does not matter much), e.g.:
    ```ql
    import java

    from Element e
    where e.fromSource()
    select e
    ```
3. Execute the test, either by running `codeql test run` or by using the VSCode extension
    :x: The test succeeds (or fails due to missing `.expected` file), but does not fail due to a Java compilation error

**Comment by aschackmull** [comment]
This has been fixed recently and should be included in the next CLI version.

**Comment by Marcono1234** [comment]
Thanks! Looks like this is indeed fixed for CodeQL CLI v2.5.0.
